flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationMasterRunner.java

```diff
@@ -597,6 +610,11 @@ public static ContainerLaunchContext createTaskManagerContext(
 
 		containerEnv.put(YarnConfigKeys.ENV_CLIENT_USERNAME, yarnClientUsername);
 
+		final String secureCookie = ENV.get(YarnConfigKeys.ENV_SECURE_AUTH_COOKIE);
+		if(secureCookie != null) {
+			containerEnv.put(YarnConfigKeys.ENV_SECURE_AUTH_COOKIE, secureCookie);
```

Comment:
The problem here is that the secure cookie will be put into the environment of the TaskManager JVM, so it'll be quite easy to just read the environment variables (not sure if that is an issue).
Another issue is that YARN is by default launching processes by creating a temporary bash file, with all the environment variables and the JVM invocation. So the secure cookie will be written into some tmp directory on YARN.
I wonder if there's some infrastructure in YARN to transfer the tokens in a secure way.